client/types.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	"context"
+
+	"github.com/tigrisdata/tigrisdb-client-go/driver"
+)
+
+// TransactionFunc is a user-provided function that will be run
+// within the context of a tranaction.
+type TransactionFunc func(
+	ctx context.Context,
+	tr driver.Tx,
+) (interface{}, error)
+
+// Client is the interface for a TigrisDB client.
+type Client interface {
+	// Database returns a Database interface for the provided
+	// database name.
+	//
+	// TODO: Return bool/error if the database exists or not
+	//       or should the client just get the error when they
+	//       try to use it? Latter is more performant.
+	Database(name string) Database
+
+	// CreateDatabaseIfNotExist creates a database if it doesn't
+	// already exist.
+	CreateDatabaseIfNotExist(
+		ctx context.Context,
+		db string,
+		options ...*driver.DatabaseOptions,
+	) error
+
+	// Driver returns the lower-level Driver interface
+	// in case the caller needs it for low-level operations.
+	Driver() driver.Driver
```

Comment:
let's not expose it here, this may confuse the user with the intention of this package.

Reply:
The reason I want to expose it is so that the client doesnt have to figure out how to construct a driver manually if they realize they need the driver api for some small thing. I can remove it if you feel strongly though

Reply:
so if a user wants to use driver interface then probably we let them import that package for now let's avoid it here.